### PR TITLE
feat(Carta Giovani Nazionale): [#176443163] New bonuses available visibility and start cgn flow logic

### PR DESCRIPTION
**Pull request by CrisTofani**
# THIS PR IS BASED ON #2718

## Short description
This PR introduces a new visibility handler logic for bonuses and adds the base saga logic to begin the cgn request flow.

## List of changes proposed in this pull request
- Removes the locale fallback for bonuses available
- Removes the API call to previous version fo the file
- Implements base saga logic for the activation flow.

## How to test
Generate Content specs based on [this](https://raw.githubusercontent.com/pagopa/io-services-metadata/2aef54d6de00b079a4549f53808fe73649a781d0/definitions.yml) file and use `io-dev-server` with `support-cgn` branch

### Before merging this PR, the merge of the PR on [io-dev-server](https://github.com/pagopa/io-services-metadata/pull/215) is needed

### CODE REVIEW

File: ts/features/bonus/bpd/saga/orchestration/onboarding/startOnboarding.ts [hunk 1]
```diff
@@ -106,7 +106,7 @@ export function* bpdStartOnboardingWorker() {
  */
 export function* handleBpdStartOnboardingSaga(): SagaIterator {
   const { cancelAction } = yield race({
-    onboarding: call(bpdStartOnboardingWorker),
+    activation: call(bpdStartOnboardingWorker),
```
Comment (Undermaken) [inline ts/features/bonus/bpd/saga/orchestration/onboarding/startOnboarding.ts:109]: is this rename needed?

File: ts/features/bonus/bonusVacanze/store/reducers/__tests__/availableBonusesTypes.test.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      {
+        ...mockBonus,
+        visibility: BonusVisibilityEnum.hidden
+      }
+    ];
+    expect(
+      visibleAvailableBonusSelector.resultFunc(pot.some(bonuses)).length
+    ).toBe(2);
+  });
+});
```
Comment (Undermaken) [inline ts/features/bonus/bonusVacanze/store/reducers/__tests__/availableBonusesTypes.test.ts:79]: I added and run another 2 tests 👯‍♀️ 
Maybe they can be added?

Note: current config mock is useless since the tests doesn't include `cgn` bonus and the FF is `YES` instead of `true`. Here the fixed one that supports the other 2 tests
```typescript

jest.mock("../../../../../../config", () => ({
  cgnEnabled: true,
  bonusVacanzeEnabled: true
}));

[...]

it("should return the experimental bonus where the FF is ON", () => {
    const visibility = BonusVisibilityEnum.experimental;
    const bonuses: BonusesAvailable = [
      { ...mockBonus, id_type: ID_BONUS_VACANZE_TYPE, visibility },
      { ...mockBonus, id_type: ID_BONUS_VACANZE_TYPE, visibility },
      { ...mockBonus, id_type: ID_BPD_TYPE, visibility }
    ];
    const result = visibleAvailableBonusSelector.resultFunc(pot.some(bonuses));
    expect(result.length).toBe(2);
    expect(result).toEqual([bonuses[0], bonuses[1]]);
  });

  it("should return the experimental bonus where the FF is ON", () => {
    const visibility = BonusVisibilityEnum.experimental;
    const bonuses: BonusesAvailable = [
      {
        ...mockBonus,
        id_type: ID_BONUS_VACANZE_TYPE,
        visibility
      },
      {
        ...mockBonus,
        id_type: ID_BPD_TYPE,
        visibility
      },
      {
        ...mockBonus,
        id_type: ID_CGN_TYPE,
        visibility
      },
      {
        ...mockBonus,
        id_type: -1,
        visibility
      }
    ];
    const result = visibleAvailableBonusSelector.resultFunc(pot.some(bonuses));
    expect(result.length).toBe(2);
    expect(result).toEqual([bonuses[0], bonuses[2]]);
  });
```

File: ts/features/bonus/cgn/store/sagas/activation/activationSaga.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  cgnActivationCancel,
+  cgnActivationStatus
+} from "../../actions/activation";
+import { navigateToCgnOnboardingInformationTos } from "../../../navigation/actions";
+
+export function* cgnStartActivationWorker() {
+  yield put(navigateToCgnOnboardingInformationTos());
+  yield put(navigationHistoryPop(1));
+
+  // HERE WILL BE IMPLEMENTED THE ACTIVATION LOGIC
```
Comment (Undermaken) [inline ts/features/bonus/cgn/store/sagas/activation/activationSaga.ts:15]:
```suggestion
  // TODO :HERE WILL BE IMPLEMENTED THE ACTIVATION LOGIC
```

File: ts/features/bonus/cgn/store/sagas/index.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import { SagaIterator } from "redux-saga";
```
Comment (Undermaken) [inline ts/features/bonus/cgn/store/sagas/index.ts:1]: In BPD feature we prefer this folders structure:

- feature
    - saga
    - store

instead of

- feature
    - store
        - saga

I find the first as a better solution since a saga is about _networking_, _orchestration_ or _navigation_. It is no strictly related to the store

What do you think?

**Review by Undermaken — CHANGES_REQUESTED** [review]

File: ts/features/bonus/bpd/saga/orchestration/onboarding/startOnboarding.ts (hunk 1 above)
Comment (CrisTofani) [reply ts/features/bonus/bpd/saga/orchestration/onboarding/startOnboarding.ts:109]: I found it more relevant calling it `activation`, since we are activating a card not properly requesting an onboarding, what do you think?

Comment (CrisTofani) [reply ts/features/bonus/bpd/saga/orchestration/onboarding/startOnboarding.ts:109]: Sorry i noticed now it was a rename on BPD saga, no it is not...IDE just exceeded in refactoring 😄

File: ts/features/bonus/cgn/store/sagas/index.ts (hunk 4 above)
Comment (CrisTofani) [reply ts/features/bonus/cgn/store/sagas/index.ts]: Fixed in b3f71c1

File: ts/features/bonus/bpd/saga/orchestration/onboarding/startOnboarding.ts (hunk 1 above)
Comment (CrisTofani) [reply ts/features/bonus/bpd/saga/orchestration/onboarding/startOnboarding.ts]: Fixed on 
ce99b47

File: ts/features/bonus/bonusVacanze/mock/availableBonuses.ts [hunk 5]
```diff
@@ -92,6 +96,73 @@ export const availableBonuses: BonusesAvailable = [
...
+    cover:
+      "https://raw.githubusercontent.com/pagopa/io-services-metadata/master/bonus/vacanze/logo/logo_BonusVacanze.png",
+    sponsorship_description: "Agenzia delle Entrate",
+    sponsorship_cover:
+      "https://raw.githubusercontent.com/pagopa/io-services-metadata/master/bonus/vacanze/logo/logo_AgenziaEntrate.png"
+  },
+  {
+    id_type: ID_BPD_TYPE,
+    it: {
+      name: "Bonus Vacanze",
```
[the remark this answers is not in the document]
Comment (CrisTofani) [reply ts/features/bonus/bonusVacanze/mock/availableBonuses.ts]: Adds in 
ce99b47

File: ts/features/bonus/bonusVacanze/mock/availableBonuses.ts [hunk 6]
```diff
@@ -1,7 +1,12 @@
 import { BonusesAvailable } from "../../../../../definitions/content/BonusesAvailable";
-import { ID_BONUS_VACANZE_TYPE } from "../utils/bonus";
```
[the remark this answers is not in the document]
Comment (CrisTofani) [reply ts/features/bonus/bonusVacanze/mock/availableBonuses.ts]: ce99b47 to solve this

**Review by Undermaken — APPROVED** [review]
